From 3a00f08140646a54e0eff2ce8938bf248dad6153 Mon Sep 17 00:00:00 2001
From: John Crispin <john@phrozen.org>
Date: Wed, 7 Aug 2019 09:59:49 +0200
Subject: [PATCH] mac80211: add missing length field increment when generating Radiotap header
Git-commit: 3a00f08140646a54e0eff2ce8938bf248dad6153
Patch-mainline: v5.4-rc1
References: jsc#SLE-13430

The code generating the Tx Radiotap header when using tx_status_ext was
missing a field increment after setting the VHT bandwidth.

Fixes: 3d07ffcaf320 ("mac80211: add struct ieee80211_tx_status support to ieee80211_add_tx_radiotap_header")
Signed-off-by: John Crispin <john@phrozen.org>
Link: https://lore.kernel.org/r/20190807075949.32414-4-john@phrozen.org
Signed-off-by: Johannes Berg <johannes.berg@intel.com>
Acked-by: Takashi Iwai <tiwai@suse.de>

---
 net/mac80211/status.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/net/mac80211/status.c b/net/mac80211/status.c
index ce0c50efd804..f88f94d1f177 100644
--- a/net/mac80211/status.c
+++ b/net/mac80211/status.c
@@ -412,6 +412,7 @@ ieee80211_add_tx_radiotap_header(struct ieee80211_local *local,
 			*pos = 0;
 			break;
 		}
+		pos++;
 
 		/* u8 mcs_nss[4] */
 		*pos = (status->rate->mcs << 4) | status->rate->nss;
-- 
2.16.4

